### PR TITLE
executor: add a test case for 'tidb_constraint_check_in_place' on temporary table

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Problem Summary:

`tidb_constraint_check_in_place` is a variable that controls the index uniqueness check.
If this variable is set to false, we will delay the check to 2PC commit. We presume the index key doesn't exist, adding those key to 2PC mutations and  ask TiKV to verify that.

For the temporary table, we do not actually write to TiKV.
So I worried this variable would conflict with the temporary table feature.
It turned out that I'm wrong... the current code works well.
Anyway, I'd like to add a test to cover that.

### What is changed and how it works?

What's Changed:

Add a test case

How it Works:

### Check List 

Tests 

- Unit test
- No code

### Release note 

- No release note

### CODE REVIEW

File: executor/write_test.go [hunk 1]
```diff
@@ -2860,6 +2860,37 @@ func (s *testSuite7) TestDeferConstraintCheckForInsert(c *C) {
...
+		tk.MustExec("begin")
+		tk.MustExec("insert into t values (1, 1)")
+		_, err = tk.Exec(`insert into t values (1, 3)`)
+		c.Assert(err, NotNil)
+		tk.MustExec("insert into t values (2, 2)")
+		_, err = tk.Exec("update t set a = a + 1 where a = 1")
+		c.Assert(err, NotNil)
+		_, err = tk.Exec("insert into t values (1, 3) on duplicated key update a = a + 1")
+		c.Assert(err, NotNil)
+		tk.MustExec("rollback")
```
Comment (djshow832) [inline executor/write_test.go:2879]: Don't you say that `tidb_constraint_check_in_place` works in the commit phase? Why do you rollback the transaction?

Comment (tiancaiamao) [reply executor/write_test.go]: Done

**Review by lysu — APPROVED** [review]

**Pull request merged**